### ISSUE TITLE
HexOrderParameter take k=float, should be int and cast later

### ISSUE THREAD

**Issue by bdice** [glotzerlab/freud#100]
**[Original report](https://bitbucket.org/glotzer/freud/issue/100) by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

Even if k needs to be used as an int later, it should be cast at c++ or cython from int

**Comment by bdice** [comment]
**Original comment by Chengyu Dai (Bitbucket: [daich](https://bitbucket.org/daich), GitHub: [daich](https://github.com/daich)).**

----------------------------------------

test = HexOrderParameter(diam_cut,k=4.)

minimum reproduction code

**Comment by bdice** [comment]
**Original comment by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

1. This is not minimum reproduction code. Something like


```
#!python

from freud.order import HexOrderParameter

diam_cut=3.0
test = HexOrderParameter(diam_cut,k=4.)
```

For what it's worth, this code does not cause an error for my install of freud. What version of freud is this failing for, and can you post the traceback?

**Comment by bdice** [comment]
**Original comment by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

@daich is this still happening?

**Comment by bdice** [comment]
**Original comment by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

@daich Can you please follow up with me about this issue? The code sample above does not produce an error for me. Can you help me understand the expected behavior if it differs from what you're seeing? Thanks!

**Comment by bdice** [comment]
**Original comment by Chengyu Dai (Bitbucket: [daich](https://bitbucket.org/daich), GitHub: [daich](https://github.com/daich)).**

----------------------------------------

Hi I was unable to produce an error either. I think my point at that time was simply saying that we should throw an error when a float is provided for k, as the code says in ` freud / freud / order.pxi ` class `HexOrderParameter`:    

 .. note:: While :math:`k` is a float, this is due to its use in calculations requiring floats. Passing in non-integer values will result in undefined behavior

**Comment by bdice** [comment]
**Original comment by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

@vyasr What do you think about this? It seems to me like we should cast the argument `k` to an integer, and the return type of `getK()` should be an integer as well. I don't think there are any sensible reasons why non-integer `k` would be useful. Internally, we would convert its type back to `float` before doing computations.

**Comment by bdice** [comment]
**Original comment by Vyas Ramasubramani (Bitbucket: [vramasub](https://bitbucket.org/vramasub), GitHub: [vyasr](https://github.com/vyasr)).**

----------------------------------------

I agree, we should just be typecasting where floats are needed but the type of `k` stored internally should be an integer. We can specify that the argument should be an integer and throw an error if it isn't. If the internal type is an integer then getK will automatically return an integer, so that should be fine. People using python2 should be importing division from __future__ anyway so the division should work out fine.

**Comment by bdice** [comment]
**Original comment by Bradley Dice (Bitbucket: [bdice](https://bitbucket.org/bdice), GitHub: [bdice](https://github.com/bdice)).**

----------------------------------------

PR submitted.

**Comment by bdice** [comment]
**Original comment by Vyas Ramasubramani (Bitbucket: [vramasub](https://bitbucket.org/vramasub), GitHub: [vyasr](https://github.com/vyasr)).**

----------------------------------------

Fixed in Pull Request #131
